spec/requests/unlinking/unlink_defendant_maat_reference_v2_spec.rb

```diff
@@ -0,0 +1,297 @@
+          params:
+  end
+
+  it 'does NOT send an unlink request to CD API' do
+    expect(a_request(:patch, api_request_path)
+          .with(body: api_request_payload.to_json, headers: json_content))
+      .not_to have_been_made
+  end
+
+  it 'renders edit' do
+    expect(response).to render_template(:edit)
+  end
+
+  it 'displays error summary with other_reason_text presence error' do
+    expect(response.body).to have_tag(:div, with: { class: 'govuk-error-summary' }) do
+      with_text error_message
+    end
+  end
+end
+
+RSpec.describe 'unlink defendant maat reference', type: :request, vcr: true, stub_unlink_v2: true do
+  include RSpecHtmlMatchers
+
+  before do
+    create(:unlink_reason, code: 1, description: 'Reason not requiring text', text_required: false)
+    create(:unlink_reason, code: 7, description: 'Reason requiring text', text_required: true)
+    allow(Feature).to receive(:enabled?).with(:laa_references).and_return(true)
+  end
+
+  let(:user) { create(:user) }
+  let(:case_urn) { 'TEST12345' }
+  let(:defendant_fixture) { load_json_stub('linked/defendant_by_reference_body.json') }
+  let(:defendant_by_id_fixture) { load_json_stub('linked_defendant.json') }
+  let(:json_api_content) { { 'Content-Type' => 'application/vnd.api+json' } }
+  let(:plain_content) { { 'Content-Type' => 'text/plain; charset=utf-8' } }
+  let(:json_content) { { 'Content-Type' => 'application/json' } }
+  let(:defendant_asn_from_fixture) { '0TSQT1LMI7CR' }
+  let(:defendant_id) { '41fcb1cd-516e-438e-887a-5987d92ef90f' }
+  let(:prosecution_case_reference_from_fixture) { 'TEST12345' }
+  let(:api_url_v2) { CdApi::BaseModel.site }
+  let(:maat_reference) { 2_123_456 }
+
+  let(:params) do
+    {
+      unlink_attempt:
+      {
+        reason_code: '1',
+        other_reason_text: ''
+      }
+    }
+  end
+  let(:api_request_path) { "#{api_url_v2}laa_references/#{defendant_id}/" }
+  let(:api_request_payload) do
+    {
+      defendant_id:,
+      user_name: user.username,
+      unlink_reason_code: 1,
+      maat_reference:
+    }
+  end
+  let(:maat_invalid_request) do
+    {
+      # rubocop:disable Style/FormatStringToken
+      title: 'The link to the court data source could not be removed. %{error_messages}',
+      # rubocop:enable Style/FormatStringToken
+      message: 'If this problem persists, please contact the IT Helpdesk on 0800 9175148.'
+    }
+  end
+  let(:maat_invalid_username) do
+    {
+      title: 'Unable to unlink this defendant',
+      message: 'User name must not exceed 10 characters'
+    }
+  end
+
+  context 'when authenticated' do
+    before do
+      sign_in user
+
+      stub_request(:get, "#{api_url}/prosecution_cases")
+        .with(query:)
+        .to_return(body: defendant_fixture, headers: json_api_content)
+
+      stub_request(:get, "#{api_url}/defendants/#{defendant_id}?include=offences")
+        .to_return(body: defendant_by_id_fixture, headers: json_api_content)
```

Comment:
I don't think these stub_requests are needed because you have stub_unlink_v2: true